### ISSUE TITLE
`from_matrix` returns the identity on a rotation matrix

### ISSUE THREAD

**Issue by cpg314** [dimforge/nalgebra#1078]
Consider the following simple program ([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=bd92af4961f4c660020c2958e9151045)), creating a rotation matrix, and creating a rotation from it.
```rust
use nalgebra; // 0.30.1

fn main() {
    let mat = nalgebra::Matrix3::<f64>::new(0.0, 0.0, 1.0, 0.0, -1.0, 0.0, 1.0, 0.0, 0.0);
    println!("{}", mat);

    // A. Returns the rotation matrix
    let mat2 = nalgebra::Rotation3::from_matrix_unchecked(mat.clone());
    println!("{}", mat2);

    // B. Returns identity
    let mat2 = nalgebra::Rotation3::from_matrix(&mat);
    println!("{}", mat2);

    // C. from_matrix_eps, starting at the identity matrix (equivalent to B.)
    // Returns identity
    let mat2 =
        nalgebra::Rotation3::from_matrix_eps(&mat, f64::EPSILON, 0, nalgebra::Rotation3::identity());
    println!("{}", mat2);

    // D. from_matrix_eps, starting away from the identity matrix
    // Returns the rotation matrix
    let rot = nalgebra::UnitQuaternion::<f64>::from_euler_angles(0.1, 0.2, 0.3);
    let mat2 =
        nalgebra::Rotation3::from_matrix_eps(&mat, f64::EPSILON, 0, rot.to_rotation_matrix());
    println!("{}", mat2);
}
```
- The use of `from_matrix` (calling `from_matrix_eps` behind the scenes) return the identity matrix instead of the original rotation matrix (B, C)
- The problem disappears when starting `from_matrix_eps` from a non-identity matrix (A, D).

Looking at the [implementation](https://github.com/dimforge/nalgebra/blob/319597e137f4a9a0f82f6570ec2964aa15c5c80d/src/geometry/rotation_specialization.rs#L728) of `SimdRealField::from_matrix_eps`, it exits after one iteration: the `axis` is the zero vector, `axisangle` is zero. Indeed, the three components of the sum
```rust
let axis = rot.column(0).cross(&m.column(0))
                + rot.column(1).cross(&m.column(1))
                + rot.column(2).cross(&m.column(2));
```
are (0,-1,0), 0 and (0,1,0).

Is this a bug or the expected behavior?

See also #661 #627

**Comment by cpg314** [comment]
Looking at https://animation.rwth-aachen.de/media/papers/2016-MIG-StableRotation.pdf we have $\tau=(0,-1,0)+0+(0,1,0)=0$. According to Appendix A, this is equivalent to minimizing _or_ maximizing the Frobenius norm, but it looks like we're hitting the maximum here given that the norm can be made zero...
From the paper:
![image](https://user-images.githubusercontent.com/44120267/154220793-5ec85880-75cf-4131-8607-4eeb5ac64968.png)
So it seems this check is necessary.

**Comment by timethy** [comment]
This is fixed with #1101
